docker/lib/dependabot/docker/file_fetcher.rb

```diff
@@ -40,63 +42,72 @@ def fetch_files
         elsif incorrectly_encoded_dockerfiles.none?
           raise(
             Dependabot::DependencyFileNotParseable,
-            incorrectly_encoded_yamlfiles.first.path
+            T.must(incorrectly_encoded_yamlfiles.first).path
           )
         else
           raise(
             Dependabot::DependencyFileNotParseable,
-            incorrectly_encoded_dockerfiles.first.path
+            T.must(incorrectly_encoded_dockerfiles.first).path
           )
         end
       end
 
       private
 
+      sig { returns(T::Array[DependencyFile]) }
       def dockerfiles
-        @dockerfiles ||=
+        @dockerfiles ||= T.let([], T.nilable(T::Array[T.untyped]))
+        @dockerfiles +=
           repo_contents(raise_errors: false)
           .select { |f| f.type == "file" && f.name.match?(DOCKER_REGEXP) }
           .map { |f| fetch_file_from_host(f.name) }
       end
 
+      sig { returns(T::Array[DependencyFile]) }
       def correctly_encoded_dockerfiles
-        dockerfiles.select { |f| f.content.valid_encoding? }
+        dockerfiles.select { |f| T.must(f.content).valid_encoding? }
```

Comment:
Does `f.content&.valid_encoding?` work here?

Reply:
This is fixed as per review comment